File: study/lua/src/app/scene/view/settlement/SummaryNormalWin.lua
local SummaryBase = require("app.scene.view.settlement.SummaryBase")
local SummaryNormalWin = class("SummaryNormalWin", SummaryBase)

local ComponentStarRule = require("app.scene.view.settlement.ComponentStarRule")
local ComponentTwoItems = require("app.scene.view.settlement.ComponentTwoItems")
local ComponentLevel = require("app.scene.view.settlement.ComponentLevel")
local ComponentLine = require("app.scene.view.settlement.ComponentLine")
local ComponentDrop = require("app.scene.view.settlement.ComponentDrop")

local TypeConvertHelper = require("app.utils.TypeConvertHelper")
local DataConst = require("app.const.DataConst")

function SummaryNormalWin:ctor(battleData, callback)
    self._battleData = battleData
    self._star = battleData.star
    self._movingName = "moving_win_"..self._star

    local list = {}
    -- local width = math.min(1136, display.width)
    -- local height = math.min(640, display.height)

    local size = G_ResolutionManager:getDesignCCSize()
    local width = size.width
    local height = size.height

    -- local midXPos = width*0.5
    local midXPos = SummaryBase.NORMAL_FIX_X

    local componentStarRule = ComponentStarRule.new(battleData.star, cc.p(midXPos, 315 - height*0.5))
    -- self:addChild(componentStarRule)
    table.insert(list, componentStarRule)

    local itemExp = 
    {
        type = TypeConvertHelper.TYPE_RESOURCE, 
        value = DataConst.RES_EXP, 
        size = self._battleData.exp,
    } 
    local itemMoney = 
    {
        type = TypeConvertHelper.TYPE_RESOURCE, 
        value = DataConst.RES_GOLD, 
        size = self._battleData.money,
    }      
    local items = {itemExp, itemMoney}
    local componentTwoItems = ComponentTwoItems.new(items, self._battleData.addAwards, cc.p(midXPos, 250 - height*0.5))
    table.insert(list, componentTwoItems)

    local componentLevel = ComponentLevel.new(self._battleData.exp, cc.p(midXPos, 220 - height*0.5))
    table.insert(list, componentLevel)

    local componentLine = ComponentLine.new("txt_sys_reward01", cc.p(midXPos, 177 - height*0.5))
    table.insert(list, componentLine)

    local componentDrop = ComponentDrop.new(self._battleData.awards, cc.p(midXPos, 107 - height*0.5), self._battleData.isDouble)
    table.insert(list, componentDrop)
    
    
    SummaryNormalWin.super.ctor(self,battleData, callback, list, midXPos, true)
end

function SummaryNormalWin:onEnter()
    SummaryNormalWin.super.onEnter(self)
    self:_createAnimation()
    --G_SignalManager:dispatch(SignalConst.EVENT_TUTORIAL_TOUCH_AUTH_BEGIN)
end

function SummaryNormalWin:onExit()
    SummaryNormalWin.super.onExit(self)
    --G_SignalManager:dispatch(SignalConst.EVENT_TUTORIAL_TOUCH_AUTH_END)
end


function SummaryNormalWin:_createAnimation()
    local effect = G_EffectGfxMgr:createPlayMovingGfx( self, self._movingName, handler(self, self._playWinText), handler(self, self.checkStart) , false )
end

return SummaryNormalWin

File: study/ts/script/scene/view/settlement/SummaryNormalWin.ts
import SummaryBase from "./SummaryBase";
import { G_ResolutionManager, G_EffectGfxMgr } from "../../../init";
import ComponentStarRule from "./ComponentStarRule";
import { TypeConvertHelper } from "../../../utils/TypeConvertHelper";
import { DataConst } from "../../../const/DataConst";
import ComponentTwoItems from "./ComponentTwoItems";
import ComponentLevel from "./ComponentLevel";
import ComponentLine from "./ComponentLine";
import ComponentDrop from "./ComponentDrop";
import { handler } from "../../../utils/handler";
import ComponentBase from "./ComponentBase";

const { ccclass, property } = cc._decorator;

@ccclass
export default class SummaryNormalWin extends SummaryBase {

    private _battleData;
    private _star: number;
    private _movingName: string;

    public init(battleData, callback) {
        this._battleData = battleData;
        this._star = battleData.star;
        this._movingName = 'moving_win_' + this._star;
        var list: ComponentBase[] = [];
        var size = G_ResolutionManager.getDesignCCSize();
        var width = size.width;
        var height = size.height;
        var midXPos = SummaryBase.NORMAL_FIX_X;
        var componentStarRule = new cc.Node("ComponentStarRule").addComponent(ComponentStarRule);
        componentStarRule.init(battleData.star, cc.v2(midXPos, 315 - height * 0.5))
        list.push(componentStarRule);
        var itemExp = {
            type: TypeConvertHelper.TYPE_RESOURCE,
            value: DataConst.RES_EXP,
            size: this._battleData.exp
        };
        var itemMoney = {
            type: TypeConvertHelper.TYPE_RESOURCE,
            value: DataConst.RES_GOLD,
            size: this._battleData.money
        };
        var items = [
            itemExp,
            itemMoney
        ];
        var componentTwoItems = new cc.Node("ComponentTwoItems").addComponent(ComponentTwoItems);
        componentTwoItems.init(items, this._battleData.addAwards, cc.v2(midXPos, 250 - height * 0.5));
        list.push(componentTwoItems);
        var componentLevel = new cc.Node("ComponentLevel").addComponent(ComponentLevel);
        componentLevel.init(this._battleData.exp, cc.v2(midXPos, 220 - height * 0.5));
        list.push(componentLevel);
        var componentLine = new cc.Node("ComponentLine").addComponent(ComponentLine)
        componentLine.init('txt_sys_reward01', cc.v2(midXPos, 177 - height * 0.5));
        list.push(componentLine);
        var componentDrop = new cc.Node("ComponentDrop").addComponent(ComponentDrop)
        componentDrop.init(this._battleData.awards, cc.v2(midXPos, 107 - height * 0.5), this._battleData.isDouble);
        list.push(componentDrop);
        super.init(battleData, callback, list, midXPos, true);
    }

    onEnter() {
        this._createAnimation();
    }

    onExit() {
        super.onExit();
    }
    _createAnimation() {
        G_EffectGfxMgr.createPlayMovingGfx(this.node, this._movingName, handler(this, this.playWinText),
            handler(this, this.checkStart), false);
    }
}